samples/drawing/drawing.cpp

```diff
@@ -574,38 +574,40 @@ void MyCanvas::DrawTestPoly(wxDC& dc)
     wxBrush brushHatch(*wxRED, wxBRUSHSTYLE_FDIAGONAL_HATCH);
     dc.SetBrush(brushHatch);
 
+    double scale = dc.GetDPIScaleFactor();
+
     wxPoint star[5];
-    star[0] = wxPoint(100, 60);
-    star[1] = wxPoint(60, 150);
-    star[2] = wxPoint(160, 100);
-    star[3] = wxPoint(40, 100);
-    star[4] = wxPoint(140, 150);
+    star[0] = scale * wxPoint(100, 60);
```

Comment:
Maybe we need `wxDC::FromDIP()` too?

Reply:
Yes, I can add that.

Reply:
I'm going to merge it like this for now, but `FromDIP()` could still be added later in another PR.

Reply:
Please wait. I noticed macOS has wrong scaling when using `wxGraphicsContext` instead of `wxDC` in the drawing sample, so `GetDPIScaleFactor` doesn't return `1.0` here. But I haven't had time to look into it.

Reply:
Oops, sorry, I hadn't noticed this before pushing :-(

I don't think this is bad enough to revert the merge, but please let me know if you think it is.

Reply:
The fix is to override `GetDPI` in `wxMacCoreGraphicsRenderer` and return `72` there. Because currently it uses the common `wxGraphicsContext::GetDPI` code that returns the DPI of the associated `wxWindow`.

I'll create a PR in the coming days if no one beats me to it.

Reply:
I could do this, but I'm not sure how exactly to test it, so maybe it would be better to wait until you can do it. But I'll create an issue for 3.1.7 for this.

Reply:
> The fix is to override `GetDPI` in `wxMacCoreGraphicsRenderer` and return `72` there. Because currently it uses the common `wxGraphicsContext::GetDPI` code that returns the DPI of the associated `wxWindow`.

No,  this is not the right fix. DPIScaleFactor is only needed for physical pixels, but we need logical pixels, and on macOS these do not correspond to physical pixel. We need FromDIP for points we use for drawing.

Reply:
I was already doubting this in #22333, and thought of another (also faulty?) solution. Using `FromDIP` for pixel sizes is advised in [`wxWindow::GetDPIScaleFactor`](https://docs.wxwidgets.org/trunk/classwx_window.html#a43bfd56d59b728157f95c838f981872c) so this indeed should also be used here.

I will compare `GetDPIScaleFactor`, `FromDIP` and `GetDPI` in `wxWindow` and `wxDC`/`wxGCDC`/`wxGraphicsContext` and make them behave the same.

Reply:
Thanks, that's perfect

Reply:
I created #22346 with these changes